heartbeat/autodiscover/builder/hints/heartbeats.go

```diff
@@ -0,0 +1,211 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package hints
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/autodiscover"
+	"github.com/elastic/beats/libbeat/autodiscover/builder"
+	"github.com/elastic/beats/libbeat/autodiscover/template"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/bus"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+func init() {
+	autodiscover.Registry.AddBuilder("hints", NewHeartbeatHints)
+}
+
+const (
+	montype    = "type"
+	schedule   = "schedule"
+	hosts      = "hosts"
+	processors = "processors"
+)
+
+type heartBeat struct {
+	config *config
+	logger *logp.Logger
+}
+
+// NewLogHints builds a log hints builder
+func NewHeartbeatHints(cfg *common.Config) (autodiscover.Builder, error) {
+	config := defaultConfig()
+	err := cfg.Unpack(config)
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to unpack hints config due to error: %v", err)
+	}
+
+	return &heartBeat{config, logp.NewLogger("hints.builder")}, nil
+}
+
+// Create config based on input hints in the bus event
+func (l *heartBeat) CreateConfig(event bus.Event) []*common.Config {
```

Comment:
I'm curious as to the choice of the variable name `l` for `*heartBeat`. Why not just `hb` or `h`.

Reply:
lol, i took the logs hints builder as a base and worked on this. copy paste